src/dmd/compiler.d

```diff
@@ -142,24 +142,34 @@ struct Compiler
             assert(0, "Unsupported source type");
         }
 
+        UnionExp ue = void;
```

Comment:
Is it possible to declare `ue` as `scope` to achieve the same result?

Reply:
What I mean is `scope` classes are allocated on the stack, so there would be no heap allocation.

Reply:
??

Reply:
Because of NRVO, ue actually gets allocated in the caller's stack.

Reply:
Nevermind.  Perhaps it wouldn't work since you're returning it and `scope` would prevent it from escaping.

Reply:
UnionExp is a struct anyway, so no need.